### PR TITLE
fix completion for set properties

### PR DESCRIPTION
**Pull request by donandren**
fix completion for set properties, now we shouldn't list get properties when set are needed

### CODE REVIEW

File: src/Avalonia.Ide.CompletionEngine.DnlibMetadataProvider/Wrappers.cs [hunk 1]
```diff
@@ -52,8 +52,8 @@ class TypeWrapper : ITypeInformation
         public IEnumerable<IPropertyInformation> Properties => _type.Properties
             //Filter indexer properties
             .Where(p =>
-                (p.GetMethod == null || p.GetMethod.Parameters.Count == (p.GetMethod.IsStatic ? 0 : 1))
```
Comment (kekekeks) [inline src/Avalonia.Ide.CompletionEngine.DnlibMetadataProvider/Wrappers.cs]: This effectively hides `internal` properties

Comment (donandren) [reply src/Avalonia.Ide.CompletionEngine.DnlibMetadataProvider/Wrappers.cs]: that's correct and intentional only public properties are to be completed, no private or internal. That's also the way wpf designer code completion works for me.

Comment (kekekeks) [reply src/Avalonia.Ide.CompletionEngine.DnlibMetadataProvider/Wrappers.cs]: It should be perfectly valid to access `internal` properties in the same assembly though.

File: src/Avalonia.Ide.CompletionEngine/AssemblyMetadata/MetadataConverter.cs [hunk 2]
```diff
@@ -102,6 +104,34 @@ public static Metadata ConvertMetadata(IMetadataReaderSession provider)
 
                         avaresValues.Add(avres);
                     }
+
+                    //add x:Class avalonia resources, they're not in the CompiledAvaloniaXaml.!AvaloniaResources format
+                    //but they are in the NamespaceInfo:
+                    foreach (var nestedType in resType.NestedTypes.Where(t => t.Name.StartsWith("NamespaceInfo:")))
```
Comment (kekekeks) [inline src/Avalonia.Ide.CompletionEngine/AssemblyMetadata/MetadataConverter.cs]: I don't think that referencing resources this way will work in run time

Comment (donandren) [reply src/Avalonia.Ide.CompletionEngine/AssemblyMetadata/MetadataConverter.cs]: we don't have general support for all types of resources anyway and that's not possible with reflection only with `avares://` way of work by design.

Here the attempt is at least to have some completion for xaml resources like Style and Styles (*.xaml) compiled with x:Class

File: src/Avalonia.Ide.CompletionEngine/AssemblyMetadata/MetadataConverter.cs [hunk 3]
```diff
@@ -187,6 +173,121 @@ public static Metadata ConvertMetadata(IMetadataReaderSession provider)
...
+                        {
+                            var assetDoc = XDocument.Load(ms);
+                            var ns = assetDoc.Root.GetDefaultNamespace();
+                            var avaResEntries = assetDoc.Root.Element(ns.GetName("Entries")).Elements(ns.GetName("AvaloniaResourcesIndexEntry"))
+                                .Select(entry => new
+                                {
+                                    Path = entry.Element(ns.GetName("Path")).Value,
+                                    Offset = int.Parse(entry.Element(ns.GetName("Offset")).Value),
+                                    Size = int.Parse(entry.Element(ns.GetName("Size")).Value)
+                                }).ToArray();
```
Comment (donandren) [inline src/Avalonia.Ide.CompletionEngine/AssemblyMetadata/MetadataConverter.cs]: @kekekeks  added some simple avares resources parsing from `!AvaloniaResources` working good

**Pull request merged**